src/sdk/pynni/nni/_graph_utils.py

```diff
@@ -510,6 +518,65 @@ def _build_index(self, nodes_op):
                 output_to_node[output] = node
         return name_to_node, input_to_node, output_to_node
 
+    def _is_key_func(self, node_cpp):
+        """
+        Judge if a cpp node is a key function node.
+        If so, we should not merge this node into the
+        adjacent node.
+        """
+        if node_cpp.kind().startswith('aten::'):
+            # the nodes that start with 'aten' are key function
+            # nodes
+            return True
+        if node_cpp.kind() in [LIST_UNPACK_KIND, TUPLE_UNPACK_KIND]:
+            # We cannot merge the List/Tuple
+            # Construct/Unpack func into other nodes, else it
+            # may lead to a graph construction error.
+            return True
+        return False
+
+    def unpack_manually(self):
+        """
+        Unpack the tensor tuple or tensor list manually,
+        and remove the ListUnpack/TupleUnpack node from
+        the graph. Note: this function will change the
+        graph structure.
+        """
+        if hasattr(self, 'unpacked'):
+            # if already unpacked the tuple/list manually
+            return
+        for node in self.nodes_py.nodes_op:
+            if node.op_type in [TUPLE_UNPACK_KIND, LIST_UNPACK_KIND]:
+                unpack_cpp = node.key_node
+                last_cpp = list(unpack_cpp.inputs())[0].node()
+                if last_cpp.kind() in [TUPLE_CONSTRUCT_KIND, LIST_CONSTRUCT_KIND]:
+                    # we need check if the tensor tuple or tensor list is produced
+                    # by a list/tuple construct node. If so, we can unpack the tuple
+                    # or list manunally.
+                    _logger.debug('List/Tuple Construct Node(cpp) %s', str(last_cpp))
+                    _logger.debug('List/Tuple Unpack Node(cpp) %s', str(unpack_cpp))
+                    assert len(list(unpack_cpp.outputs())) == len(list(last_cpp.inputs()))
```

Comment:
a little confused about this assert, i think the main reason is i don't understand what is `last_cpp`